x-pack/plugins/apm/server/routes/storage_explorer/get_service_statistics.ts

```diff
@@ -0,0 +1,240 @@
+  indexLifeCyclePhaseToDataTier,
+} from '../../../common/storage_explorer_types';
+import { getTotalTransactionsPerService } from './get_total_transactions_per_service';
+import {
+  PROCESSOR_EVENT,
+  SERVICE_NAME,
+  SERVICE_ENVIRONMENT,
+  TIER,
+  TRANSACTION_SAMPLED,
+  AGENT_NAME,
+  INDEX,
+} from '../../../common/elasticsearch_fieldnames';
+import { environmentQuery } from '../../../common/utils/environment_query';
+import { AgentName } from '../../../typings/es_schemas/ui/fields/agent';
+import {
+  getTotalIndicesStats,
+  getEstimatedSizeForDocumentsInIndex,
+} from './indices_stats_helpers';
+import { RandomSampler } from '../../lib/helpers/get_random_sampler';
+
+async function getMainServiceStatistics({
+  setup,
+  context,
+  indexLifecyclePhase,
+  randomSampler,
+  start,
+  end,
+  environment,
+  kuery,
+}: {
+  setup: Setup;
+  context: ApmPluginRequestHandlerContext;
+  indexLifecyclePhase: IndexLifecyclePhaseSelectOption;
+  randomSampler: RandomSampler;
+  start: number;
+  end: number;
+  environment: string;
+  kuery: string;
+}) {
+  const { apmEventClient } = setup;
+
+  const allIndicesStats = await getTotalIndicesStats({ context, setup });
+
+  const response = await apmEventClient.search('get_main_service_statistics', {
+    apm: {
+      events: [
+        ProcessorEvent.span,
+        ProcessorEvent.transaction,
+        ProcessorEvent.error,
+        ProcessorEvent.metric,
+      ],
+    },
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            ...environmentQuery(environment),
+            ...kqlQuery(kuery),
+            ...rangeQuery(start, end),
+            ...(indexLifecyclePhase !== IndexLifecyclePhaseSelectOption.All
+              ? termQuery(
+                  TIER,
+                  indexLifeCyclePhaseToDataTier[indexLifecyclePhase]
+                )
+              : []),
+          ] as QueryDslQueryContainer[],
+        },
+      },
+      aggs: {
+        sample: {
+          random_sampler: randomSampler,
+          aggs: {
+            services: {
+              terms: {
+                field: SERVICE_NAME,
+                size: 500,
+              },
+              aggs: {
+                sample: {
+                  top_hits: {
+                    size: 1,
+                    _source: [AGENT_NAME],
+                    sort: {
+                      '@timestamp': 'desc',
+                    },
+                  },
+                },
+                indices: {
+                  terms: {
+                    field: INDEX,
+                    size: 500,
+                  },
+                  aggs: {
+                    number_of_metric_docs: {
+                      value_count: {
+                        field: INDEX,
+                      },
+                    },
+                  },
+                },
+                environments: {
+                  terms: {
+                    field: SERVICE_ENVIRONMENT,
```

Comment:
should you add `missing` here?

Reply:
Good shout, but it seems that it's not there in service inventory for [transaction stats](https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/server/routes/services/get_services/get_service_transaction_stats.ts#L111) and [error / metrics](https://github.com/elastic/kibana/blob/a771b021e111543396a8b3a417583a25ebefdb60/x-pack/plugins/apm/server/routes/services/get_services/get_services_from_error_and_metric_documents.ts#L73). Am I missing something? Do you think it should be added there as well?

Reply:
I don't think so - those are aggs to list the environments for a service. We don't show NOT_DEFINED as a tag I think.

Reply:
That's why I asked this agg is used for the same reason, to display the list of environments for each service in the storage explorer table